blog/content/post/2021-12-14-secure-jtag-on-imx-soc.md

```diff
@@ -0,0 +1,280 @@
+keys. We can get challenge key during the first try of connection. Let's try
+```
+$ sudo openocd -f interface/ftdi/olimex-arm-usb-ocd-h.cfg -f target/imx6.cfg
+Open On-Chip Debugger 0.9.0-dev-00019-g970a12aef-dirty (2021-10-11-14:19)
+Licensed under GNU GPL v2
+For bug reports, read
+    http://openocd.sourceforge.net/doc/doxygen/bugs.html
+Info : only one transport option; autoselect 'jtag'
+Warn : imx6.sdma: nonstandard IR value
+adapter speed: 10 kHz
+Info : clock speed 10 kHz
+SJC  : Using database tcl/target/imx6_sjcauth.txt
+SJC  : Challenge 0x0e81d4eacb3785
+SJC  : Response  0x574c20308fad77
+SJC  : Using database tcl/target/imx6_sjcauth.txt
+SJC  : Challenge 0x0e81d4eacb3785
+SJC  : Response  0x574c20308fad77
+SJC  : Using database tcl/target/imx6_sjcauth.txt
+SJC  : Challenge 0x0e81d4eacb3785
+SJC  : Response  0x574c20308fad77
+Info : JTAG tap: imx6.dap tap/device found: 0x4ba00477 (mfg: 0x23b, part: 0xba00
+, ver: 0x4)
+Info : TAP imx6.sdma does not have IDCODE
+Info : JTAG tap: imx6.sjc tap/device found: 0x1891a01d (mfg: 0x00e, part: 0x891a
+, ver: 0x1)
+Info : imx6.cpu.0: hardware has 6 breakpoints, 4 watchpoints
+Info : number of cache level 1
+Info : imx6.cpu.0 cluster 0 core 0 multi core
+```
+
+Works fine, now a response key is required to access JTAG. It looks like we can
+lock this mode and key overriding by burning relevant fuses
+```
+# ./crucible -m IMX6UL -r 1 -b 16 -e big blow JTAG_SMODE 0x3
+# ./crucible -m IMX6UL -r 1 -b 16 -e big blow BOOT_CFG_LOCK 0x3
+```
+
+It is time to verify our solution - remove line with key from `imx6_sjcauth.txt`
+and try to connect again:
+```
+$ sudo openocd -f interface/ftdi/olimex-arm-usb-ocd-h.cfg -f target/imx6.cfg
+Open On-Chip Debugger 0.9.0-dev-00019-g970a12aef-dirty (2021-10-11-14:19)
+Licensed under GNU GPL v2
+For bug reports, read
+	http://openocd.sourceforge.net/doc/doxygen/bugs.html
+Info : only one transport option; autoselect 'jtag'
+Warn : imx6.sdma: nonstandard IR value
+adapter speed: 10 kHz
+Info : clock speed 10 kHz
+SJC  : Using database tcl/target/imx6_sjcauth.txt
+SJC  : Challenge 0x0e81d4eacb3785
+SJC  : Device not in database, not authenticating
+SJC  : Using database tcl/target/imx6_sjcauth.txt
+SJC  : Challenge 0x0e81d4eacb3785
+SJC  : Device not in database, not authenticating
+SJC  : Using database tcl/target/imx6_sjcauth.txt
+SJC  : Challenge 0x0e81d4eacb3785
+SJC  : Device not in database, not authenticating
+Error: JTAG scan chain interrogation failed: all ones
+Error: Check JTAG interface, timings, target power, etc.
+Error: Trying to use configured scan chain anyway...
+Error: imx6.dap: IR capture error; saw 0x0f not 0x01
+SJC  : Using database tcl/target/imx6_sjcauth.txt
+SJC  : Challenge 0x0e81d4eacb3785
+SJC  : Device not in database, not authenticating
+Warn : Bypassing JTAG setup events due to errors
+Warn : Invalid ACK 0x7 in JTAG-DP transaction
+```
+
+As you can see, JTAG access is disabled without response key.
```

Comment:
I can't see because I'm new to this and it is hard to me to read so complicated output. Can you explain in blog post what I should look for and how to interpret the output.

Reply:
Changed